### ISSUE TITLE
Android Crash - NetworkOnMainThreadException

### ISSUE THREAD

**Issue by bFlood** [eclipsesource/tabris-js#656]
I've only seen this crash once but I figured I'd pass it along

```log
java.lang.RuntimeException: Unable to destroy activity {com.sdlportal.sdlmobile3/com.eclipsesource.tabris.android.TabrisActivity}: android.os.NetworkOnMainThreadException
	at android.app.ActivityThread.performDestroyActivity(ActivityThread.java:4109)
	at android.app.ActivityThread.handleDestroyActivity(ActivityThread.java:4127)
	at android.app.ActivityThread.access$1500(ActivityThread.java:172)
	at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1475)
	at android.os.Handler.dispatchMessage(Handler.java:102)
	at android.os.Looper.loop(Looper.java:145)
	at android.app.ActivityThread.main(ActivityThread.java:5835)
	at java.lang.reflect.Method.invoke(Native Method)
	at java.lang.reflect.Method.invoke(Method.java:372)
	at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:1399)
	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1194)
Caused by: android.os.NetworkOnMainThreadException
	at android.os.StrictMode$AndroidBlockGuardPolicy.onNetwork(StrictMode.java:1147)
	at com.android.org.conscrypt.OpenSSLSocketImpl$SSLOutputStream.write(OpenSSLSocketImpl.java:778)
	at okio.Okio$1.write(Okio.java:78)
	at okio.AsyncTimeout$1.write(AsyncTimeout.java:155)
	at okio.RealBufferedSink.flush(RealBufferedSink.java:185)
	at com.squareup.okhttp.internal.spdy.Spdy3$Writer.rstStream(Spdy3.java:382)
	at com.squareup.okhttp.internal.spdy.SpdyConnection.writeSynReset(SpdyConnection.java:351)
	at com.squareup.okhttp.internal.spdy.SpdyStream.close(SpdyStream.java:222)
	at com.squareup.okhttp.internal.http.SpdyTransport.disconnect(SpdyTransport.java:215)
	at com.squareup.okhttp.internal.http.HttpEngine.disconnect(HttpEngine.java:468)
	at com.squareup.okhttp.Call.cancel(Call.java:118)
	at com.eclipsesource.tabris.android.internal.toolkit.view.HttpRequest.cancel(HttpRequest.java:114)
	at com.eclipsesource.tabris.android.internal.toolkit.operator.HttpRequestOperator.destroy(HttpRequestOperator.java:136)
	at com.eclipsesource.tabris.android.internal.toolkit.AndroidWidgetToolkit.destroyObjects(AndroidWidgetToolkit.java:266)
	at com.eclipsesource.tabris.android.internal.toolkit.AndroidWidgetToolkit.dispose(AndroidWidgetToolkit.java:233)
	at com.eclipsesource.tabris.android.TabrisActivity.onDestroy(TabrisActivity.java:237)
	at android.app.Activity.performDestroy(Activity.java:6451)
	at android.app.Instrumentation.callActivityOnDestroy(Instrumentation.java:1154)
	at android.app.ActivityThread.performDestroyActivity(ActivityThread.java:4096)
```

**Comment by mpost** [comment]
Thanks for reporting... We have seen the issue before but couldn't pinpoint it. We will have another look.

**Comment by jonek** [comment]
@bFlood Do you remember the situation when you observed this issue? Any details could help.

**Comment by jonek** [comment]
I tried to reproduce the problem when I recognized that you stacktrace indicates you were trying to cancle a spdy http connection. To reproduce the problem I only used the node http-server which did not gave me a spdy http connection but a plain http connection.
Obviously we found this issue https://github.com/square/retrofit/issues/1069 indicating a general problem on Android when spdy http connection are canceled.

**Comment by bFlood** [comment]
@jonek no I don't, sorry. it was an automated report from the Google Play store and the tester did not write any comments on what he was doing when it crashed

**Comment by mpost** [comment]
Here is the direct issue on the okhttp issue tracker: https://github.com/square/okhttp/issues/1592

**Comment by goranch** [comment]
I can reproduce it every time when I change the log level from RestAdapter.LogLevel.FULL to BASIC or remove the logLevel altogether

RestAdapter.Builder builder = new RestAdapter.Builder()
                .setEndpoint(baseUrl)
               //.setLogLevel(RestAdapter.LogLevel.FULL).setLog(new AndroidLog(TAG))
                .setClient(new OkClient(new OkHttpClient()));

**Comment by mpost** [comment]
The issue has been fixed in the upcoming okhttp 2.7 release. we will update asap.

**Comment by mpost** [comment]
This issue has been fixed by updating to okhttp 2.7.2.
